File: core/src/main/java/pcb/uwu/units/derived/electromagnetism/ElectricPotentialUnit.java
package pcb.uwu.units.derived.electromagnetism;

import pcb.uwu.core.CompositeUnit;
import pcb.uwu.core.UnitCounter;
import pcb.uwu.units.base.ElectricCurrentUnit;
import pcb.uwu.units.base.LengthUnit;
import pcb.uwu.units.base.MassUnit;
import pcb.uwu.units.base.TimeUnit;

public class ElectricPotentialUnit extends CompositeUnit {

	// region base Constructor

	public ElectricPotentialUnit(MassUnit massUnit, LengthUnit lengthUnit, ElectricCurrentUnit electricCurrentUnit, TimeUnit timeUnit) {
		super(new UnitCounter()
				.major(massUnit)
				.major(lengthUnit, 2)
				.minor(electricCurrentUnit)
				.minor(timeUnit, 3));
	}

	// endregion

}

File: dimension/src/main/java/pcb/uwu/unit/derived/electromagnetism/ElectricPotentialUnit.kt
package pcb.uwu.unit.derived.electromagnetism

import pcb.uwu.core.CompositeUnit
import pcb.uwu.core.UnitCounter
import pcb.uwu.unit.base.ElectricCurrentUnit
import pcb.uwu.unit.base.LengthUnit
import pcb.uwu.unit.base.MassUnit
import pcb.uwu.unit.base.TimeUnit
import pcb.uwu.unit.derived.termodynamics.EnergyUnit
import pcb.uwu.unit.derived.termodynamics.PowerUnit

open class ElectricPotentialUnit : CompositeUnit
{
    constructor(massUnit: MassUnit,
                lengthUnit: LengthUnit,
                electricCurrentUnit: ElectricCurrentUnit,
                timeUnit: TimeUnit)
            : super(UnitCounter()
                        .major(massUnit)
                        .major(lengthUnit)
                        .major(lengthUnit)
                        .minor(electricCurrentUnit)
                        .minor(timeUnit)
                        .minor(timeUnit)
                        .minor(timeUnit))

    constructor(energyUnit: EnergyUnit,
                electricChargeUnit: ElectricChargeUnit)
            : super(UnitCounter()
                        .major(energyUnit)
                        .minor(electricChargeUnit))

    constructor(powerUnit: PowerUnit,
                electricCurrentUnit: ElectricCurrentUnit)
            : super(UnitCounter()
                        .major(powerUnit)
                        .minor(electricCurrentUnit))

    constructor(electricChargeUnit: ElectricChargeUnit,
                electricCapacitanceUnit: ElectricCapacitanceUnit)
            : super(UnitCounter()
                        .major(electricChargeUnit)
                        .minor(electricCapacitanceUnit))

    constructor(magneticFluxUnit: MagneticFluxUnit,
                timeUnit: TimeUnit)
            : super(UnitCounter()
                        .major(magneticFluxUnit)
                        .minor(timeUnit))

    constructor(electricResistanceUnit: ElectricResistanceUnit,
                electricCurrentUnit: ElectricCurrentUnit)
            : super(UnitCounter()
                        .major(electricResistanceUnit)
                        .major(electricCurrentUnit))

    constructor(electricCurrentUnit: ElectricCurrentUnit,
                electricConductanceUnit: ElectricConductanceUnit)
            : super(UnitCounter()
                        .major(electricCurrentUnit)
                        .major(electricConductanceUnit))
}
